appside/frames/consent_disclosure.js

```diff
@@ -0,0 +1,137 @@
+"use strict";
+
+/**
+ * Rendering (View) code for consent disclosure frame
+ * @author Rachel Sitt
+ */
+
+
+/**
+ * A frame specifically purposed for consent disclosure.
+ * 
+ * Frame contains pdf file and a set of checkboxes.
+ **/
+class ConsentDisclosureFrame extends Frame {
+
+    /**
+     * Construct ConsentDisclosureFrame from an object
+     * 
+     * @param frame_data -- Object containing the frame's data. Expected fields:
+     *    frame_data.template -- The frame's template
+     *    frame_data.title (string) -- The frame's title
+     *    frame_data.instructions (string) -- instructions for user to read pdf
+     *    frame_data.questions (Array of map of string) -- 2 key/value pairs
+     *    -- formatted as (key: question, value: boolean); value is false by default
+     *  Behavior undefined if frame does not have these properties.
+     */
+    constructor(frame_data) {
+        super();
+        
+        this.template = frame_data.template;
+        this.title = frame_data.title;
+        this.instructions = frame_data.instructions;
+        this.questions = frame_data.questions;
+        this.user_input = new Map();
+    }
+
+    /**
+     * Render this frame into the DOM
+     * 
+     * @require -- DOM must have a div whose ID is 'frame'
+     * 
+     * @effects -- Does not preserve former content of <div id="frame">.
+     *    Renders the data from this into that div.
+     */
+    render() {
+
+        // make a new empty div with id frame, not yet in the dom
+        let frame = document.createElement('div'); 
+        $(frame).attr('id', 'frame');
+        
+        // insert a h2 node for the title
+        let title = document.createElement('h5');
+        $(title).text(this.title);
+        $(title).attr('class', 'text-info text-uppercase mb-2');
+        frame.appendChild(title);
+
+        let instructions = document.createElement('h5');
+        $(instructions).attr('class', 'text-info text-uppercase mb-2');
+
+        $(instructions).attr('class', 'consent_instructions');
+        $(instructions).text(this.instructions);
+        frame.appendChild(instructions);
+
+        let pdf = document.createElement('a');
+        $(pdf).attr('href', 'images/consent.pdf');
+        $(pdf).attr('target', '_blank');    // opens pdf in new window/tab
+        $(pdf).attr('class', 'consent_pdf');
+        $(pdf).text('Consent Disclosure Form');
+        frame.appendChild(pdf);
+
+        let container = document.createElement('div');  // flexbox for content
+        $(container).attr('class', 'consent_frame');
+
+        $(pdf).click(function() {   // ONLY when pdf is clicked will checkboxes appear
```

Comment:
ooh neat! I like that a change happens when the link is clicked. 

Instead of making them _appear_, could we start by having them grayed out, then have them un-gray when the link is clicked? It could be confusing to the participant to see the form link but nothing to do afterward.

Reply:
done :)